diff --git a/src/cmd/vendor/github.com/google/pprof/internal/binutils/addr2liner.go b/src/cmd/vendor/github.com/google/pprof/internal/binutils/addr2liner.go
index c0661bf..d8841d2 100644
--- a/src/cmd/vendor/github.com/google/pprof/internal/binutils/addr2liner.go
+++ b/src/cmd/vendor/github.com/google/pprof/internal/binutils/addr2liner.go
@@ -18,6 +18,7 @@ import (
 	"bufio"
 	"fmt"
 	"io"
+ 	"os"
 	"os/exec"
 	"strconv"
 	"strings"
@@ -85,7 +86,11 @@ func (a *addr2LinerJob) close() {
 // program under consideration.
 func newAddr2Liner(cmd, file string, base uint64) (*addr2Liner, error) {
 	if cmd == "" {
-		cmd = defaultAddr2line
+		if env_cmd, ok := os.LookupEnv("ADDR2LINE"); ok {
+			cmd = env_cmd
+		} else {
+			cmd = defaultAddr2line
+		}
 	}
 
 	j := &addr2LinerJob{
