From 1614a3a6ad53f4ca7de53ad4fd6b55b7898d6f60 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Patrick=20Sodr=C3=A9?= <psodre@gmail.com>
Date: Wed, 8 Apr 2020 23:52:24 -0400
Subject: [PATCH 1/5] issue10607 is only valid when running with CGO

---
 test/fixedbugs/issue10607.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/test/fixedbugs/issue10607.go b/test/fixedbugs/issue10607.go
index 448a37dcac..92f9dc5053 100644
--- a/test/fixedbugs/issue10607.go
+++ b/test/fixedbugs/issue10607.go
@@ -1,4 +1,4 @@
-// +build linux,!ppc64,!riscv64,gc
+// +build linux,!ppc64,!riscv64,gc,cgo
 // run
 
 // Copyright 2015 The Go Authors. All rights reserved.
-- 
2.30.2

