From bcc04311d9e9353330da5ad3d639779c2eeb8fd0 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Patrick=20Sodr=C3=A9?= <psodre@gmail.com>
Date: Sat, 11 Jun 2022 08:47:11 -0400
Subject: [PATCH 2/7] Add CC,{C,CPP,LD}FLAGS to script_test extraEnvKeys

---
 src/cmd/go/script_test.go | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/cmd/go/script_test.go b/src/cmd/go/script_test.go
index 624c5bf..2db02af 100644
--- a/src/cmd/go/script_test.go
+++ b/src/cmd/go/script_test.go
@@ -282,9 +282,14 @@ var extraEnvKeys = []string{
 	"LIBRARY_PATH",       // allow override of non-standard static library paths
 	"C_INCLUDE_PATH",     // allow override non-standard include paths
 	"CC",                 // don't lose user settings when invoking cgo
+	"CFLAGS",             // conda: don't lose compiler settings when invoking cgo
+	"CONDA_BUILD_SYSROOT",// conda: don't lose CONDA_BUILD_SYSROOT when invoking cgo
+	"CPPFLAGS",           // conda: don't lose compiler settings when invoking cgo
 	"GO_TESTING_GOTOOLS", // for gccgo testing
 	"GCCGO",              // for gccgo testing
 	"GCCGOTOOLDIR",       // for gccgo testing
+	"LDFLAGS",            // conda: don't lose compiler settings when invoking cgo
+	"MACOSX_DEPLOYMENT_TARGET",  // conda: don't lose the deployment target when invoking cgo
 }
 
 // updateSum runs 'go mod tidy', 'go list -mod=mod -m all', or
