From f19847d2719610e7e3ae4e5306e10186848f3c6b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Patrick=20Sodr=C3=A9?= <psodre@gmail.com>
Date: Tue, 7 Apr 2020 23:02:19 -0400
Subject: [PATCH 3/6] Fix cgo_fortran test setup for conda

---
 misc/cgo/fortran/test.bash | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/misc/cgo/fortran/test.bash b/misc/cgo/fortran/test.bash
index 2b61730815..9e8948bae7 100755
--- a/misc/cgo/fortran/test.bash
+++ b/misc/cgo/fortran/test.bash
@@ -20,11 +20,7 @@ fi
 
 case "$FC" in
 *gfortran*)
-  libpath=$(dirname $($FC -print-file-name=libgfortran.$libext))
-  if [ "$goos" != "aix" ]; then
-	  RPATH_FLAG="-Wl,-rpath,$libpath"
-  fi
-  export CGO_LDFLAGS="$CGO_LDFLAGS $RPATH_FLAG -L $libpath"
+  # CONDA's gfortran + go does not need changes to CGO_LDFLAGS
   ;;
 esac
 
-- 
2.24.2 (Apple Git-127)

