From 12120ea1dd6c259cd540ebb68b7cd977a995d8ff Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Patrick=20Sodr=C3=A9?= <psodre@gmail.com>
Date: Tue, 7 Apr 2020 21:20:31 -0400
Subject: [PATCH 3/6] Add CC,{C,CPP,LD}FLAGS to script_test extraEnvKeys

---
 src/cmd/go/script_test.go | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/cmd/go/script_test.go b/src/cmd/go/script_test.go
index ec498bbcd7..d42d9d1569 100644
--- a/src/cmd/go/script_test.go
+++ b/src/cmd/go/script_test.go
@@ -90,9 +90,14 @@ var extraEnvKeys = []string{
 	"WINDIR",             // must be preserved on Windows to be able to run PowerShell command; golang.org/issue/30711
 	"LD_LIBRARY_PATH",    // must be preserved on Unix systems to find shared libraries
 	"CC",                 // don't lose user settings when invoking cgo
+	"CFLAGS",             // conda: don't lose compiler settings when invoking cgo
+	"CONDA_BUILD_SYSROOT",// conda: don't lose CONDA_BUILD_SYSROOT when invoking cgo
+	"CPPFLAGS",           // conda: don't lose compiler settings when invoking cgo
 	"GO_TESTING_GOTOOLS", // for gccgo testing
 	"GCCGO",              // for gccgo testing
 	"GCCGOTOOLDIR",       // for gccgo testing
+	"LDFLAGS",            // conda: don't lose compiler settings when invoking cgo
+	"MACOSX_DEPLOYMENT_TARGET",  // conda: don't lose the deployment target when invoking cgo
 }
 
 // setup sets up the test execution temporary directory and environment.
-- 
2.30.0

