From 80697ab8da802a73a7e5c4f9c7d17a442efe594b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Patrick=20Sodr=C3=A9?= <psodre@gmail.com>
Date: Tue, 7 Apr 2020 21:20:31 -0400
Subject: [PATCH 4/5] Add CC,{C,CPP,LD}FLAGS to script_test extraEnvKeys

---
 src/cmd/go/script_test.go | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/cmd/go/script_test.go b/src/cmd/go/script_test.go
index c56c1fd3e4..5b6345ebe9 100644
--- a/src/cmd/go/script_test.go
+++ b/src/cmd/go/script_test.go
@@ -84,6 +84,12 @@ type backgroundCmd struct {
 
 var extraEnvKeys = []string{
 	"SYSTEMROOT", // must be preserved on Windows to find DLLs; golang.org/issue/25210
+	"CC",
+	"CFLAGS",
+	"CPPFLAGS",
+	"LDFLAGS",
+	"CONDA_BUILD_SYSROOT",
+	"MACOSX_DEPLOYMENT_TARGET",
 }
 
 // setup sets up the test execution temporary directory and environment.
-- 
2.26.0

