From 3427337eb85c055cdb2e24340bb0169b68901fac Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Patrick=20Sodr=C3=A9?= <psodre@gmail.com>
Date: Sat, 18 Apr 2020 22:18:19 -0400
Subject: [PATCH 6/6] Change default GOPATH to $SRC_DIR/gopath

---
 src/go/build/build.go | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/go/build/build.go b/src/go/build/build.go
index f74efb2852..58a6f6d5ae 100644
--- a/src/go/build/build.go
+++ b/src/go/build/build.go
@@ -264,7 +264,10 @@ var Default Context = defaultContext()
 
 func defaultCondaGOPATH() string {
 	if envIsOne("CONDA_GO_COMPILER") && envIsOne("CONDA_BUILD") {
-		return envOr("SRC_DIR", defaultGOPATH())
+		if srcDir := os.Getenv("SRC_DIR"); srcDir != "" {
+			return filepath.Join(srcDir, "gopath")
+		}
+		return defaultGOPATH()
 	}
 	return defaultGOPATH()
 }
@@ -348,7 +351,6 @@ func envIsOne(name string) bool {
 	return s == "1"
 }
 
-
 // An ImportMode controls the behavior of the Import method.
 type ImportMode uint
 
-- 
2.26.0

