From 2138e46f86bef3072cb052aa28dbc40cf2431bec Mon Sep 17 00:00:00 2001
From: Nehal J Wani <nehaljw.kkd1@gmail.com>
Date: Mon, 25 Jan 2021 11:28:34 -0500
Subject: [PATCH 7/7] Disable TestPIESize if CGO isn't enabled

---
 src/cmd/link/elf_test.go | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/cmd/link/elf_test.go b/src/cmd/link/elf_test.go
index 88048ed2c5..4a57668271 100644
--- a/src/cmd/link/elf_test.go
+++ b/src/cmd/link/elf_test.go
@@ -226,6 +226,8 @@ func main() {
 
 func TestPIESize(t *testing.T) {
 	testenv.MustHaveGoBuild(t)
+	testenv.MustHaveCGO(t)
+
 	if !sys.BuildModeSupported(runtime.Compiler, "pie", runtime.GOOS, runtime.GOARCH) {
 		t.Skip("-buildmode=pie not supported")
 	}
-- 
2.30.0

