--- a/src/cmd/link/internal/ld/pe.go
+++ b/src/cmd/link/internal/ld/pe.go
@@ -662,7 +662,7 @@
 	}
 }
 
-// addInitArray adds .ctors COFF section to the file f.
+// addInitArray adds the runtime initializer COFF section to the file f.
 func (f *peFile) addInitArray(ctxt *Link) *peSection {
 	// The size below was determined by the specification for array relocations,
 	// and by observing what GCC writes here. If the initarray section grows to
@@ -678,7 +678,12 @@
 		size = 4
 		alignment = IMAGE_SCN_ALIGN_4BYTES
 	}
-	sect := f.addSection(".ctors", size, size)
+	name := ".ctors"
+	if ctxt.BuildMode == BuildModeCShared && ctxt.isMSVC() {
+		// Register the Go initializer with the MSVC CRT.
+		name = ".CRT$XCU"
+	}
+	sect := f.addSection(name, size, size)
 	sect.characteristics = IMAGE_SCN_CNT_INITIALIZED_DATA | IMAGE_SCN_MEM_READ | IMAGE_SCN_MEM_WRITE | alignment
 	sect.sizeOfRawData = uint32(size)
 	ctxt.Out.SeekSet(int64(sect.pointerToRawData))
