diff --git a/misc/cgo/fortran/test.bash b/misc/cgo/fortran/test.bash
index 1e0d59ea1c..58a8753883 100755
--- a/misc/cgo/fortran/test.bash
+++ b/misc/cgo/fortran/test.bash
@@ -18,8 +18,7 @@ fi
 
 case "$FC" in
 *gfortran*)
-  libpath=$(dirname $($FC -print-file-name=libgfortran.$libext))
-  export CGO_LDFLAGS="$CGO_LDFLAGS -Wl,-rpath,$libpath -L $libpath"
+  # CONDA's gfortran + go does not need changes to CGO_LDFLAGS
   ;;
 esac
 
diff --git a/src/cmd/go/go_test.go b/src/cmd/go/go_test.go
index 1c3dcd851b..ab482c04ef 100644
--- a/src/cmd/go/go_test.go
+++ b/src/cmd/go/go_test.go
@@ -1580,6 +1580,35 @@ func TestInstallIntoGOBIN(t *testing.T) {
 	tg.wantExecutable("testdata/bin1/go-cmd-test"+exeSuffix, "go install go-cmd-test did not write to testdata/bin1/go-cmd-test")
 }
 
+// Test that without $GOBIN set, binaries get installed
+// into the PREFIX/bin directory when CONDA_BUILD is 1.
+func TestInstallIntoGOBINWithCondaCompiler(t *testing.T) {
+	tg := testgo(t)
+	defer tg.cleanup()
+	tg.setenv("CONDA_GO_COMPILER", "1")
+	prefix := filepath.Join(tg.pwd(), "testdata", "conda_prefix")
+	tg.creatingTemp(prefix)
+	tg.setenv("CONDA_PREFIX", prefix)
+	tg.setenv("GOPATH", filepath.Join(tg.pwd(), "testdata"))
+	tg.run("install", "go-cmd-test")
+	tg.wantExecutable("testdata/conda_prefix/bin/go-cmd-test"+exeSuffix, "go install go-cmd-test did not write to testdata/conda_prefix/bin/go-cmd-test")
+}
+
+// Test that without $GOBIN set, binaries get installed
+// into the PREFIX/bin directory when CONDA_BUILD is 1.
+func TestInstallIntoGOBINWithCondaBuild(t *testing.T) {
+	tg := testgo(t)
+	defer tg.cleanup()
+	tg.setenv("CONDA_GO_COMPILER", "1")
+	tg.setenv("CONDA_BUILD", "1")
+	prefix := filepath.Join(tg.pwd(), "testdata", "conda_build_prefix")
+	tg.creatingTemp(prefix)
+	tg.setenv("PREFIX", prefix)
+	tg.setenv("SRCDIR", filepath.Join(tg.pwd(), "testdata"))
+	tg.run("install", "go-cmd-test")
+	tg.wantExecutable("testdata/conda_build_prefix/bin/go-cmd-test"+exeSuffix, "go install go-cmd-test did not write to testdata/conda_build_prefix/bin/go-cmd-test")
+}
+
 // Issue 11065
 func TestInstallToCurrentDirectoryCreatesExecutable(t *testing.T) {
 	tg := testgo(t)
diff --git a/src/cmd/go/internal/cfg/cfg.go b/src/cmd/go/internal/cfg/cfg.go
index 8dc4d1fbd2..c63ef675f5 100644
--- a/src/cmd/go/internal/cfg/cfg.go
+++ b/src/cmd/go/internal/cfg/cfg.go
@@ -94,7 +94,7 @@ func init() {
 
 var (
 	GOROOT       = findGOROOT()
-	GOBIN        = os.Getenv("GOBIN")
+	GOBIN        = envOr("GOBIN", defaultCondaGOBIN())
 	GOROOTbin    = filepath.Join(GOROOT, "bin")
 	GOROOTpkg    = filepath.Join(GOROOT, "pkg")
 	GOROOTsrc    = filepath.Join(GOROOT, "src")
@@ -198,3 +198,38 @@ func isGOROOT(path string) bool {
 	}
 	return stat.IsDir()
 }
+
+// defaultCondaGOBIN returns a conda-aware GOBIN path if CONDA_GO_COMPILER
+// is set to 1, otherwise it returns the default empty-string.
+// When CONDA_GO_COMPILER is set to 1, GOBIN's default value is set to:
+//    - $PREFIX/bin if CONDA_BUILD is set to 1 and the PREFIX is set
+//    - $CONDA_PREFIX/bin if CONDA_BUILD is not set to 1 and CONDA_PREFIX is set
+//    - ""  the default value
+func defaultCondaGOBIN() string {
+	if envIsOne("CONDA_GO_COMPILER") {
+		prefixEnv := "CONDA_PREFIX"
+		if envIsOne("CONDA_BUILD") {
+			prefixEnv = "PREFIX"
+		}
+		if prefix, ok := os.LookupEnv(prefixEnv); ok {
+			return filepath.Join(prefix, "bin")
+		}
+	}
+	return ""
+}
+
+func envOr(name, def string) string {
+	s := os.Getenv(name)
+	if s == "" {
+		return def
+	}
+	return s
+}
+
+// envIsOne returns true if environment variable is set to 1, it
+// it returns false otherwise
+func envIsOne(name string) bool {
+	s := os.Getenv(name)
+	return s == "1"
+}
+
diff --git a/src/cmd/go/internal/work/exec.go b/src/cmd/go/internal/work/exec.go
index 0ab7976c56..4dfa856221 100644
--- a/src/cmd/go/internal/work/exec.go
+++ b/src/cmd/go/internal/work/exec.go
@@ -2373,6 +2373,32 @@ func (b *Builder) gccArchArgs() []string {
 	return nil
 }
 
+// condaGetCompilerFlags returns the value of the given environment variable that is normally available when
+// using conda based compilers, or return the unmodified default no environment variable is found.
+// On linux, it will disable the linker's gc-sections option.
+// On mac/darwin, it will change/reset the macosx minimal version, and set the isysroot if available.
+func condaGetCompilerFlags(key, def string) string {
+	if value, ok := os.LookupEnv(key); ok {
+		switch cfg.Goos {
+		case "linux":
+			if key == "LDFLAGS" {
+				return value + " -Wl,--no-gc-sections"
+			}
+		case "darwin":
+			if key == "CPPFLAGS" {
+				if macosxDefaultTarget, ok := os.LookupEnv("MACOSX_DEFAULT_TARGET"); ok {
+					value = value + " -mmacosx-version-min=" + macosxDefaultTarget
+				}
+				if sysroot, ok := os.LookupEnv("CONDA_BUILD_SYSROOT"); ok {
+					value = value + " -isysroot " + sysroot
+				}
+			}
+		}
+		return value
+	}
+	return def
+}
+
 // envList returns the value of the given environment variable broken
 // into fields, using the default value when the variable is empty.
 func envList(key, def string) []string {
@@ -2410,7 +2436,7 @@ func buildFlags(name, defaults string, fromPackage []string, check func(string,
 	if err := check(name, "#cgo "+name, fromPackage); err != nil {
 		return nil, err
 	}
-	return str.StringList(envList("CGO_"+name, defaults), fromPackage), nil
+	return str.StringList(envList("CGO_"+name, condaGetCompilerFlags(name, defaults)), fromPackage), nil
 }
 
 var cgoRe = regexp.MustCompile(`[/\\:]`)
diff --git a/src/cmd/go/script_test.go b/src/cmd/go/script_test.go
index c56c1fd3e4..5b6345ebe9 100644
--- a/src/cmd/go/script_test.go
+++ b/src/cmd/go/script_test.go
@@ -84,6 +84,12 @@ type backgroundCmd struct {
 
 var extraEnvKeys = []string{
 	"SYSTEMROOT", // must be preserved on Windows to find DLLs; golang.org/issue/25210
+	"CC",
+	"CFLAGS",
+	"CPPFLAGS",
+	"LDFLAGS",
+	"CONDA_BUILD_SYSROOT",
+	"MACOSX_DEPLOYMENT_TARGET",
 }
 
 // setup sets up the test execution temporary directory and environment.
diff --git a/src/go/build/build.go b/src/go/build/build.go
index 0fa67201f8..f74efb2852 100644
--- a/src/go/build/build.go
+++ b/src/go/build/build.go
@@ -262,6 +262,12 @@ func (ctxt *Context) SrcDirs() []string {
 // if set, or else the compiled code's GOARCH, GOOS, and GOROOT.
 var Default Context = defaultContext()
 
+func defaultCondaGOPATH() string {
+	if envIsOne("CONDA_GO_COMPILER") && envIsOne("CONDA_BUILD") {
+		return envOr("SRC_DIR", defaultGOPATH())
+	}
+	return defaultGOPATH()
+}
 func defaultGOPATH() string {
 	env := "HOME"
 	if runtime.GOOS == "windows" {
@@ -289,7 +295,7 @@ func defaultContext() Context {
 	c.GOARCH = envOr("GOARCH", runtime.GOARCH)
 	c.GOOS = envOr("GOOS", runtime.GOOS)
 	c.GOROOT = pathpkg.Clean(runtime.GOROOT())
-	c.GOPATH = envOr("GOPATH", defaultGOPATH())
+	c.GOPATH = envOr("GOPATH", defaultCondaGOPATH())
 	c.Compiler = runtime.Compiler
 
 	// Each major Go release in the Go 1.x series should add a tag here.
@@ -335,6 +341,14 @@ func envOr(name, def string) string {
 	return s
 }
 
+// envIsOne returns true if environment variable is set to 1, it
+// it returns false otherwise
+func envIsOne(name string) bool {
+	s := os.Getenv(name)
+	return s == "1"
+}
+
+
 // An ImportMode controls the behavior of the Import method.
 type ImportMode uint
 
